### PR TITLE
Improve the Readme for the repo

### PR DESCRIPTION
**Pull request by mavasani**
1. More emphasis on Microsoft.CodeAnalysis.FxCopAnalyzers package as the primary analyzer package of the repo
2. Add pointers to docs.microsoft.com documentation for migration from legacy FxCop to FxCop analyzers.

First step towards addressing #2025.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @gewarren @jinujoseph @vatsalyaagrawal @lucabol @Andrew-Webb

File: README.md [hunk 1]
```diff
@@ -1,92 +1,93 @@
...
 
-|          |Windows Debug|Windows Release|
-|:--------:|:-----------:|:-------------:|
-|**master**|[![Build Status](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Debug/badge/icon)](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Debug/) [![codecov](https://codecov.io/gh/dotnet/roslyn-analyzers/branch/master/graph/badge.svg)](https://codecov.io/gh/dotnet/roslyn-analyzers)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Release/badge/icon)](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Release/)|
+This is the **primary analyzer package** for this repo that contains all **the ported FxCop code analysis rules (CAxxxx)**. Refer to https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers for the documentation and FAQs on migration from legacy post-build static analysis ("Analyze" -> "Run Code Analysis" command in Visual Studio), also known as "FxCop", to FxCop analyzers.
```
Comment (sharwell) [inline README.md]:
```suggestion
This is the **primary analyzer package** for this repo that contains all **the ported FxCop code analysis rules (CAxxxx)**. Refer to https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers for the documentation and FAQs on migration from legacy post-build static analysis ("Analyze" &rarr; "Run Code Analysis" command in Visual Studio), also known as "FxCop", to FxCop analyzers.
```

File: README.md [hunk 2]
```diff
@@ -1,92 +1,93 @@
-﻿.NET Compiler Platform ("Roslyn") Analyzers
+﻿.NET Compiler Platform ("Roslyn") Analyzers, including ported [FxCop analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers)
 ===========================================
 
 [![Join the chat at https://gitter.im/roslyn-analyzers/Lobby](https://badges.gitter.im/roslyn-analyzers/Lobby.svg)](https://gitter.im/roslyn-analyzers/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-This repository contains a number of [Roslyn](https://github.com/dotnet/roslyn) diagnostic analyzers initially developed to help flesh out the design and implementation of the static analysis APIs. They have been migrated from the main [dotnet/roslyn](https://github.com/dotnet/roslyn) repository in order to continue and speed their further development.
+This repository contains Roslyn [diagnostic analyzers and code fix providers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview) for general _code quality_ improvement for C# and VisualBasic projects targeting either _.NetCore/.NetStandard_ or _Desktop .NetFramework_.
```
Comment (gewarren) [inline README.md]:
```suggestion
This repository contains Roslyn [diagnostic analyzers and code fix providers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview) for general _code quality_ improvement for C# and Visual Basic projects that target either _.NET Core, .NET Standard,_ or _.NET Framework for desktop apps_.
```

File: README.md [hunk 3]
```diff
@@ -1,92 +1,93 @@
...
 
-|          |Windows Debug|Windows Release|
-|:--------:|:-----------:|:-------------:|
-|**master**|[![Build Status](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Debug/badge/icon)](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Debug/) [![codecov](https://codecov.io/gh/dotnet/roslyn-analyzers/branch/master/graph/badge.svg)](https://codecov.io/gh/dotnet/roslyn-analyzers)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Release/badge/icon)](https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/Windows_NT_Release/)|
+This is the **primary analyzer package** for this repo that contains all **the ported FxCop code analysis rules (CAxxxx)**. Refer to https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers for the documentation and FAQs on migration from legacy post-build static analysis ("Analyze" &rarr; "Run Code Analysis" command in Visual Studio), also known as "FxCop", to FxCop analyzers.
```
Comment (gewarren) [inline README.md]:
```suggestion
This is the **primary analyzer package** for this repo that contains all **the ported FxCop code analysis rules (CAxxxx)**. Refer to [FxCop analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers) for documentation and FAQs about migrating from legacy post-build static analysis ("Analyze" &rarr; "Run Code Analysis" command in Visual Studio), also known as "FxCop", to FxCop analyzers.
```

File: README.md [hunk 4]
```diff
@@ -1,92 +1,93 @@
...
 
-[//]: # (End current test results)
+This analyzer package contains all the ported FxCop rules that are applicable for both _.NetCore/.NetStandard_ and _Desktop .NetFramework_ projects. You **do not need to install any separate analyzer package from this repo to get target framework specific FxCop rules**.
```
Comment (gewarren) [inline README.md]:
```suggestion
This analyzer package contains all the ported FxCop rules that are applicable for both _.NetCore/.NetStandard_ and _Desktop .NetFramework_ projects. You **do not need to install any separate analyzer package from this repo to get target-framework specific FxCop rules**.
```

File: README.md [hunk 5]
```diff
@@ -1,92 +1,93 @@
...
 
-Projects
-========
+This analyzer package contains the following sub-packages or NuGet dependencies that are automatically installed when you install Microsoft.CodeAnalysis.FxCopAnalyzers package:
```
Comment (gewarren) [inline README.md]:
```suggestion
This analyzer package contains the following subpackages or NuGet dependencies that are automatically installed when you install the Microsoft.CodeAnalysis.FxCopAnalyzers package:
```

File: README.md [hunk 6]
```diff
@@ -1,92 +1,93 @@
...
 
-Microsoft.CodeAnalysis.Analyzers
---------------------------------
+### Microsoft.NetCore.Analyzers
 
-*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Microsoft.CodeAnalysis.Analyzers.svg)](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)
+*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Microsoft.NetCore.Analyzers.svg)](https://www.nuget.org/packages/Microsoft.NetCore.Analyzers)
 
-Provides guidelines for using .NET Compiler Platform ("Roslyn") APIs.
+This package contains rules for correct usage of APIs that are present in _.NetCore/.NetStandard_ framework libraries. For example, [CA1309](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1309-use-ordinal-stringcomparison) (UseOrdinalStringComparison) flags usages of string compare APIs that do not specify `StringComparison` argument.
```
Comment (gewarren) [inline README.md]:
```suggestion
This package contains rules for correct usage of APIs that are present in _.NetCore/.NetStandard_ framework libraries. For example, [CA1309](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1309-use-ordinal-stringcomparison) (UseOrdinalStringComparison) flags usages of string compare APIs that don't specify a `StringComparison` argument.
```

File: README.md [hunk 7]
```diff
@@ -1,92 +1,93 @@
...
 
-Provides common code quality guidelines.
+*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Microsoft.NetFramework.Analyzers.svg)](https://www.nuget.org/packages/Microsoft.NetFramework.Analyzers)
 
-[More info](src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md)
+This package contains rules for correct usage of APIs that are present only in _Desktop .NetFramework_ libraries.
 
-Microsoft.CodeQuality.Analyzers.Exp
---------------------------------
+**NOTE:** The analyzers in this package will silently bail out if installed on a _.NetCore/.NetStandard_ project that do not have the underlying API whose usage is being checked. If future versions of _.NetCore/.NetStandard_ libraries start including these APIs, the analyzers will automatically light up on _.NetCore/.NetStandard_ projects targeting these libraries.
```
Comment (gewarren) [inline README.md]:
```suggestion
**NOTE:** The analyzers in this package will silently bail out if installed on a _.NetCore/.NetStandard_ project that do not have the underlying API whose usage is being checked. If future versions of _.NetCore/.NetStandard_ libraries include these APIs, the analyzers will automatically light up on _.NetCore/.NetStandard_ projects that target these libraries.
```

File: README.md [hunk 8]
```diff
@@ -1,92 +1,93 @@
...
 
-Experimental package based on Dataflow Analysis to provide code quality guidelines in executable code.
+### Text.Analyzers
 
-[More info](src/Microsoft.CodeQuality.Analyzers.Exp/Microsoft.CodeQuality.Analyzers.Exp.md)
+*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Text.Analyzers.svg)](https://www.nuget.org/packages/Text.Analyzers)
 
-Microsoft.NetCore.Analyzers
------------------
+This package contains naming rules for identifiers and for text included in code, such as comments. Currently, this package has no implemented rules, but future versions will have some of the naming rules from https://docs.microsoft.com/en-us/visualstudio/code-quality/naming-warnings.
```
Comment (gewarren) [inline README.md]:
```suggestion
This package contains naming rules for identifiers and for text included in code, such as comments. Currently, this package has no implemented rules, but future versions will have some of the naming rules from [legacy FxCop naming warnings](https://docs.microsoft.com/en-us/visualstudio/code-quality/naming-warnings).
```

File: README.md [hunk 9]
```diff
@@ -1,92 +1,93 @@
...
 
-[More info](src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md)
+### Microsoft.CodeAnalysis.Analyzers
 
-Microsoft.NetFramework.Analyzers
------------------
+*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Microsoft.CodeAnalysis.Analyzers.svg)](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)
 
-*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Microsoft.NetFramework.Analyzers.svg)](https://www.nuget.org/packages/Microsoft.NetFramework.Analyzers)
+This package contains rules for correct usage of APIs from [Microsoft.CodeAnalysis](https://www.nuget.org/packages/Microsoft.CodeAnalysis) NuGet package, i.e. .NET Compiler Platform ("Roslyn") APIs. These are primarily aimed towards helping authors of diagnostic analyzers and code fix providers to invoke the Microsoft.CodeAnalysis APIs in a recommended manner.
```
Comment (gewarren) [inline README.md]:
```suggestion
This package contains rules for correct usage of APIs from the [Microsoft.CodeAnalysis](https://www.nuget.org/packages/Microsoft.CodeAnalysis) NuGet package, i.e. .NET Compiler Platform ("Roslyn") APIs. These are primarily aimed towards helping authors of diagnostic analyzers and code fix providers to invoke the Microsoft.CodeAnalysis APIs in a recommended manner.
```

File: README.md [hunk 10]
```diff
@@ -1,92 +1,93 @@
...
 
-Text.Analyzers
--------------------------------
 
-*Latest stable version:* [![NuGet](https://img.shields.io/nuget/v/Text.Analyzers.svg)](https://www.nuget.org/packages/Text.Analyzers)
+### MetaCompilation (prototype)
+
+*Created by summer 2015 interns [Zoë Petard](https://github.com/zoepetard), [Jessica Petty](https://github.com/jepetty), and [Daniel King](https://github.com/daking2014)*
+
+The MetaCompilation Analyzer is an analyzer that functions as a tutorial to teach users how to write an analyzer. It uses diagnostics and code fixes to guide the user through the various steps required to create a simple analyzer. It is designed for a novice analyzer programmer with some previous programming experience.
```
Comment (gewarren) [inline README.md]:
```suggestion
The MetaCompilation Analyzer is an analyzer that functions as a tutorial to teach users how to write an analyzer. It uses diagnostics and code fixes to guide the user through the various steps required to create a simple analyzer. It is designed for novice analyzer developers who have some previous programming experience.
```

**Comment by mavasani** [comment]
@gewarren Thanks for the feedback. Does this look good to merge?

**Pull request merged**